Dfe.Academies.Academisation.Service/Mappers/Legacy/ApplicationAggregate/LegacyApplicationServiceModelMapper.cs

```diff
@@ -0,0 +1,103 @@
+﻿using Dfe.Academies.Academisation.Domain.Core.ApplicationAggregate;
+using Dfe.Academies.Academisation.IDomain.ApplicationAggregate;
+using Dfe.Academies.Academisation.IService.ServiceModels.Legacy.ApplicationAggregate;
+
+namespace Dfe.Academies.Academisation.Service.Mappers.Legacy.ApplicationAggregate;
+
+internal class LegacyApplicationServiceModelMapper
+{
+	internal static LegacyApplicationServiceModel MapToLegacyServiceModel(IApplication application)
+	{
+		LegacyApplicationServiceModel serviceModel = new(
+			LegacyKeyPersonServiceModelMapper.MapToServiceModels(application),
+			LegacySchoolServiceModelMapper.MapToServiceModels(application)
+			)
+		{
+			ApplicationId = application.ApplicationId.ToString(),
+
+			// For now this legacy service model only supports JoinAMat
+			ApplicationType = MapType(application),
+			ApplicationSubmitted = application.ApplicationStatus == ApplicationStatus.Submitted,
+
+			// These don't seem to be set in the Azure Data Factory pipeline
+			ApplicationStatusId = null,
+
+			// ToDo: ApplicationReference
+			////Name = application.ApplicationReference,
+		};
+
+		// How do we determine the Lead Author, we have a list of contributors?
+		IContributor? leadAuthor = application.Contributors.FirstOrDefault();
+
+		if (leadAuthor is not null)
+		{
+			serviceModel = serviceModel with
+			{
+				ApplicationLeadAuthorId = leadAuthor.Id.ToString(),
+				ApplicationLeadEmail = leadAuthor.Details.EmailAddress,
+				ApplicationVersion = null,
+				ApplicationLeadAuthorName = $"{leadAuthor.Details.FirstName} {leadAuthor.Details.LastName}",
+				ApplicationRole = null,
+				ApplicationRoleOtherDescription = null,
+			};
+		}
+
+		if (application.ApplicationType == ApplicationType.JoinAMat)
+		{
+			serviceModel = serviceModel with
+			{
+				// I think these will come from application.ExistingTrust but we don't have that yet
+				TrustName = null,
+				TrustId = null,
+				ChangesToTrust = null,
+				ChangesToTrustExplained = null,
+				ChangesToLaGovernance = null,
+				ChangesToLaGovernanceExplained = null,
+				TrustApproverName = null,
+				TrustApproverEmail = null,
+			};
+		}
+		else
+		{
+			// For FormAMat and FormASat it would be necessary to set these fields
+			// (but it's probably not worth implementing them in this legacy service model)
+			serviceModel = serviceModel with
```

Comment:
Why set all these individually to null? Couldn't the serviceModel be made null?

Reply:
There are lots of other fields in this ServiceModel that aren't null. Put this code here to show what field would need to be set if we implement FormAMat in the LegacyApplicationServiceModel (which I suspect we won't)

Reply:
Makes sense. Thanks